_sources/functions_in_cpp/mixed_up_code.rst

```diff
@@ -0,0 +1,344 @@
+         return 'B'  #paired
+   =====
+      }
+   =====
+      else if (percentage >= 70) {
+   =====
+      else if (percentage > 70) {  #paired
+   =====
+         return 'C';
+   =====
+      }
+   =====
+      else {
+   =====
+         return 'F';
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_4
+   :numbered: left
+   :adaptive:
+   :noindent:
+   :practice: T
+
+   Let's write the code for the triangleArea function. triangleArea
+   takes two parameters, base and height. It returns the 
+   area of the triangle using the formula 1/2 * base * height.
+   Put the necessary blocks of code in the correct order.
+   -----
+   double triangleArea (double base, double height) {
+   =====
+   int triangleArea (double base, double height) {  #distractor
+   =====
+   void triangleArea (double base, double height) {  #distractor
+   =====
+   double triangleArea (base, height) {  #distractor
+   =====
+      double area;  #distractor
+   =====
+      return 0.5 * base * height;
+   =====
+      cout << 0.5 * base * height << endl;  #distractor
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_5
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the cylinderVolume function. cylinderVolume
+   takes two parameters, radius and height. It returns the 
+   volume of the cylinder using the formula pi * radius * radius * height.
+   Put the necessary blocks of code in the correct order.
+   -----
+   double cylinderVolume (double radius, double height) {
+   =====
+   void cylinderVolume (double radius, double height) {  #distractor
+   =====
+   double cylinderVolume (radius, height) {  #distractor
+   =====
+      double pi = 3.14;
+   =====
+      return pi * radius * radius * height;
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_6
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write a function called ``scale`` which takes an input ``x`` and scales it by a factor 
+   of 2 is ``x`` is even and a factor of 3 if ``x`` is odd. The function should modify the original
+   ``x`` variable, not return the scaled value.
+   -----
+   void scale(int &x) {
+   =====
+   void scale(int x) {  #distractor
+   =====
+   int scale(int x) {  #distractor
+   =====
+      if (x % 2 == 0) {
+   =====
+      if (x % 2) {  #paired
+   =====
+         x = x * 2;
+   =====
+      }
+   =====
+      return x * 2;
+   =====
+      else {
+   =====
+         x = x * 3;
+   =====
+         return x * 3;  #distractor
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_7
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the isDoubleDigit function. isDoubleDigit
+   takes num as a parameter. isDoubleDigit returns true if 
+   num is a double digit number and returns false otherwise. 
+   In the code, use the if then else structure. 
+   Put the necessary blocks of code in the correct order.
```

Comment:
I don't love this problem, because there is more than one way to solve it (either using an `else` block or not using an `else` block), and this forces the students to use the `else` block, which is really less elegant than the other way. I would either remove this problem, or revise it so that students don't even have an option to use an `else` block.